workflow/rules/mapping.smk

```diff
@@ -0,0 +1,69 @@
+rule bwa_index:
+    input:
+        config["resources"]["ref"]["genome"]
+    output:
+        multiext(config["resources"]["ref"]["genome"], ".amb", ".ann", ".bwt", ".pac", ".sa")
+    log:
+        "logs/bwa/bwa_index.log"
+    params:
+        algorithm="bwtsw"
+    wrapper:
+        "0.55.1/bio/bwa/index"
+
+rule bwa_mem:
+    input:
+        reads = get_map_reads_input,
+        idx = rules.bwa_index.output
+    output:
+        temp("results/mapped/{sample}-{unit}.bam")
+    log:
+        "logs/bwa/bwa_mem/{sample}-{unit}.log"
+    params:
+        index= lambda w, input: os.path.splitext(input.idx[0])[0],
+        extra= get_read_group,
+        sort="samtools",
+        sort_order="coordinate",
+    threads: 8
+    wrapper:
+        "0.52.0/bio/bwa/mem"
+
+rule mark_duplicates:
+    input:
+        "results/mapped/{sample}-{unit}.bam"
+    output:
+        bam=temp("results/mapped/dedup/{sample}-{unit}.bam"),
+        metrics="results/mapped/dedup/{sample}-{unit}.metrics.txt"
+    log:
+        "logs/picard/dedup/{sample}-{unit}.log"
+    params:
+        "REMOVE_DUPLICATES=true"
+    wrapper:
+        "0.55.1/bio/picard/markduplicates"
```

Comment:
Upon having another look at the nf-core workflow, I just realised that they only claim to do a MarkDupliates before sample merging, but the `main.nf` does not actually do this. So I think we should actually remove this step for now. So sorry for the extra hassle of removing the intermediate step and its output from the multiqc input.